assembly/src/parsers/ast/context.rs

```diff
@@ -158,8 +158,8 @@ impl ParserContext {
             let proc_name_hash = self.get_proc_name_hash(full_proc_name);
             Ok(Node::Instruction(Instruction::ExecImported(proc_name_hash)))
         } else {
-            let index = self.procedures.get(&label).unwrap().index;
-            Ok(Node::Instruction(Instruction::ExecLocal(index)))
+            let index = self.procedures.get(&label).unwrap().0;
```

Comment:
Instead of using `unwrap` here (and below as well) - shouldn't we return an error if the procedure is not found?

also: let's rename `self.procedures` into `self.local_procs` - I think this name would be more descriptive.